src/ipld.rs

```diff
@@ -94,53 +162,71 @@ impl<'a> From<&'a str> for IpldIndex<'a> {
     }
 }
 
+impl<'a> TryFrom<IpldIndex<'a>> for usize {
+    type Error = AccessError;
+
+    fn try_from(index: IpldIndex<'a>) -> Result<Self, Self::Error> {
+        let parsed = match index {
+            IpldIndex::List(i) => i,
+            IpldIndex::Map(ref key) => key
+                .parse()
+                .map_err(|_| AccessError::ParseInteger(key.to_string()))?,
```

Comment:
Why do we allow indexing arrays by integers anyways?

Reply:
And vice-versa? Honestly, that's more important because it relates to the whole "can we have integer keys" debate.

Reply:
NOTE: not relevant for _this_ PR. I can file an issue if you'd prefer to discuss there.

Reply:
Map keys are still always strings. Though I'll revisit this whole things in a subsequent PR.